src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/DataFlowExtensions/DataFlowExtensionMethodCaller.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Utilities.DataFlowExtensions
+{
+    /// <summary>
+    /// <see cref="DataFlowExtensionMethodCaller"/> is a caller of a <seealso cref="IDataFlowExtensionWrapper"/>,
+    /// which wraps a call to extension methods defined in DataFlowExtensions. These methods cannot be mocked since
+    /// these are static methods. Hence we are covering the static methods with wrapper which makes the behavior non-static
+    /// which will then let us mock
+    /// </summary>
+    public class DataFlowExtensionMethodCaller
```

Comment:
@davkean I think the behavior here is the same as using a Mock object. In the product code, we provide an actual implementation of `IDataFlowExtensionWrapper` and in the test code, we provide a Mocked `IDataFlowExtensionWrapper`. This is a similar pattern with Mock testing where we include mocked objects to provide the desired behavior which helps testing.

Also by providing the `IReceivableSourceBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>> sourceBlock` to `DataFlowExtensionMethodCaller`, we are creating a component which calls us back.